src/api/sso/test/row-level-policies.test.js

```diff
@@ -0,0 +1,348 @@
+const { hash } = require('@senecacdot/satellite');
+const { createClient } = require('@supabase/supabase-js');
+
+const { SUPABASE_URL, ANON_KEY } = process.env;
+const {
+  createTelescopeUsers,
+  cleanupTelescopeUsers,
+  addFeeds,
+  cleanFeeds,
+} = require('./supabase-util');
+const { createToken } = require('../src/token');
+
+// Regular Telescope user
+const hansLippershey = {
+  firstName: 'Hans',
+  lastName: 'Lippershey',
+  email: 'hans-lippershey@example.com',
+  displayName: 'Hans Lippershey',
+  isAdmin: false,
+  isFlagged: false,
+  feeds: ['https://imaginary.blog.com/feed/hans'],
+  githubUsername: 'hlippershey',
+  githubAvatarUrl:
+    'https://avatars.githubusercontent.com/u/33902374?s=460&u=733c50a2f50ba297ed30f6b5921a511c2f43bfee&v=4',
+};
+const johannesKepler = {
+  firstName: 'Johannes',
+  lastName: 'Kepler',
+  email: 'user1@example.com',
+  displayName: 'Johannes Kepler',
+  isAdmin: false,
+  isFlagged: false,
+  feeds: ['https://imaginary.blog.com/feed/johannes'],
+  githubUsername: 'jkepler',
+  githubAvatarUrl:
+    'https://avatars.githubusercontent.com/u/7242003?s=460&u=733c50a2f50ba297ed30f6b5921a511c2f43bfee&v=4',
+};
+
+// unregistered user
+const unregisteredUser = {
+  firstName: 'Galileo',
+  lastName: 'Galilei',
+  email: 'user2@example.com',
+  displayName: 'Galileo Galilei',
+  isAdmin: false,
+  isFlagged: false,
+  feeds: ['https://dev.to/galileogalilei'],
+  githubUsername: 'octocat',
+  githubAvatarUrl: 'https://avatars.githubusercontent.com/u/583231?v=4',
+};
+
+// Create 2 Telescope accounts in the database
+const users = [hansLippershey, johannesKepler];
+
+// Create 2 feeds, 1 belongs to a user, 1 does not belong to anyone
+const feedList = [
+  {
+    url: hansLippershey.feeds[0],
+    wikiAuthorName: hansLippershey.displayName,
+    userId: hash(hansLippershey.email),
+  },
+  {
+    url: unregisteredUser.feeds[0],
+    wikiAuthorName: unregisteredUser.displayName,
+  },
+];
+
+describe('Unauthenticated users with public anon_key', () => {
+  beforeAll(() => {
+    supabase = createClient(SUPABASE_URL, ANON_KEY);
```

Comment:
You need a `let supabase;` above this `beforeAll` so you don't create a global.  Or even better:

```js
const supabase = () => createClient(SUPABASE_URL, ANON_KEY);
...
await supabase().from(...)
```

Not sharing state between tests is ideal.

Reply:
I don't understand.  You're creating a global when you do this.  You have to use `let` or `const` somewhere.  What am I missing?

Reply:
When creating separate clients, the second initialized by an accessToken still uses the old key and makes the tests fail. The test for the authorized client only works when I called setAuth.

Reply:
What I think is Supabase doesn't expect us to create many clients, once one client is initialized, the next calls to `createClient()` returns the already initialized one vs creating a new one.

Reply:
It looks like it always makes a new one to me:

https://github.com/supabase/supabase-js/blob/master/src/index.ts#L13-L22

Reply:
I've tried a couple of things:
- Initialize clients in `beforeAll`: `const supabase = createClient(...)` -> tests fail because supabase is undefined
- Initialize at the top of each `describe` block:  Tests for the authenticated client fail, even though being initialized using an access token

Reply:
I don't think the client connects to any services, it just initializes a class used for making HTTP request

Reply:
I think I got it. `createClient` expects a public URL, and na API key, either a public anon key or service role key. Our JWT is not an API key so we can not pass it to `createClient`. The access token can only be used with `setAuth`